src/onegov/town6/api.py

```diff
@@ -215,31 +217,51 @@ def for_filter(self, **filters: Any) -> Self:
 
         """
 
-        return self.__class__(self.request, self.directory, filters)
+        return self.__class__(self.request, self.endpoint, filters)
 
     def by_id(
             self, id: PKType
               ) -> ExtendedDirectoryEntry | None:
         """ Return the item with the given ID from the collection. """
 
         try:
-            return self.__class__(self.request, self.directory
+            return self.__class__(self.request, self.endpoint
                                   ).collection.by_id(id)
         except SQLAlchemyError:
             return None
 
     def item_data(self, item: ExtendedDirectoryEntry) -> dict[str, Any]:
-        data = item.content['values'].copy() or {}
-        data['coordinates'] = get_geo_location(item)
-
-        for field in item.directory.fields:
-            if any(type in field.type for type in ['fileinput', 'url']):
-                data.pop(field.id, None)
+        data: dict[str, Any] = {}
+
+        if item.access == 'public':
+            if item.content_fields:
+                data = {f.name: f.object_data for f in item.content_fields}
+
+            for field in item.directory.fields:
+                if any(field_type in field.type for field_type in [
+                    'fileinput', 'url']):
+                    data.pop(field.id, None)
+                if any(field_type in field.type for field_type in [
+                    'date', 'datetime']):
+                    if data.get(field.id):
+                        data[field.id] = data[field.id].isoformat()
```

Comment:
Can we roll this into a single loop? You have access to the type of field in the comprehension above, filtering out the file fields can be the `if` part of the comprehension and for the value you can write a little helper function that formats `date`/`datetime` values.